src/jti_background.js

```diff
@@ -204,16 +202,33 @@ function setDefaultTemplates (callback) {
 function downloadJSONData (url, callback) {
     fetchJSON(url, function (status, message, data) {
         if (status) {
-            data.templates = JSONtoTemplateData(data.templates);
-            data.options.domains = JSONtoDomainData(data.options.domains);
-            data.options.inputIDs = JSONtoInputIDData(data.options.inputIDs);
+            JSONtoData(data);
             saveTemplates(data, callback);
         } else {
             callback(false, message);
         }
     });
 }
 
+function loadLocalFile (fileContents, callback) {
+    try {
+        var templateJSON = JSON.parse(fileContents);
+        JSONtoData(templateJSON);
+        saveTemplates(templateJSON, callback);
+    } catch (e) {
+        callback(false, 'Error parsing JSON. Please verify file contents');
+    }
+}
+
+function JSONtoData (data) {
+    data.templates = JSONtoTemplateData(data.templates);
+    if (data.options === undefined) {
+        data.options = emptyData.options;
+    }
+    data.options.domains = JSONtoDomainData(data.options.domains);
+    data.options.inputIDs = JSONtoInputIDData(data.options.inputIDs);
```

Comment:
I think we could simply do something like https://api.jquery.com/jquery.extend/
With the data and the emptyData, allowing us to eliminate all condition checks.